tests/spark/test_mrjob_spark_harness.py

```diff
@@ -488,32 +545,40 @@ def _test_shuffle_and_sort(self, sort_values):
             if called_groupBy:
                 if '.' in name:
                     continue  # Python 3.4/3.5 tracks groupBy.assert_called()
-
-                if not kwargs.get('preservesPartitioning'):
-                    import pdb; pdb.set_trace()
                 self.assertEqual(kwargs.get('preservesPartitioning'), True)
             elif name == 'groupBy':
                 called_groupBy = True
 
         # check that groupBy() was actually called
         self.assertTrue(called_groupBy)
 
-    def test_run_reducer(self):
+    def test_run_reducer_with_num_reducers(self):
+        self._test_run_reducer(num_reducers=1)
+
+    def test_run_reducer_without_num_reducers(self):
+        self._test_run_reducer()
+
+    def _test_run_reducer(self, num_reducers=None):
         rdd = self.mock_rdd()
 
         reducer_job = Mock()
         reducer_job.pick_protocols.return_value = (Mock(), Mock())
 
-        final_rdd = _run_reducer(reducer_job, rdd)
+        final_rdd = _run_reducer(reducer_job, rdd, num_reducers=num_reducers)
         self.assertEqual(final_rdd, rdd)  # mock RDD's methods return it
 
         called_mapPartitions = False
-        for name, args, kwargs in rdd.method_calls:
+        expected_preservePartitioning_vals = [
```

Comment:
I'd prefer if you didn't make the test expect exactly three calls; it should be possible to refactor `_run_reducer()` a bit without having to update this test. The test is just trying to check `groupBy()` and every call after.

Reply:
I slightly modified the call, so that we don't limit to 3  calls.